File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        stack = []
        
        for i in s:
            if i == '(' or i == '[' or i == '{':
                stack.append(i)
            else:
                
                if(len(stack) == 0):
                    return False

                popped = stack.pop()

                if(popped == '(' and i != ')' or 
                   popped == '[' and i != ']' or 
                   popped == '{' and i != '}'):
                    
                    return False
        
        if len(stack) == 0:
            return True
        else:
            return False

File: 0020-valid-parentheses/0020-valid-parentheses.java
import java.util.Stack;

class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack1 = new Stack<>();
        
        for (char c : s.toCharArray()) {
        	
        	if (c == '(' || c == '{' || c == '[') {
        		stack1.add(c);
        	}
        	else {
        		if(stack1.isEmpty()) {
        			return false;
        		}
        		
                char popped_element = stack1.pop();
        		if(popped_element == '{' && c != '}' ||
        				popped_element == '[' && c != ']'||
        					popped_element == '(' && c != ')') {
        			
        			return false;
        		}
        	}
        }
        return stack1.isEmpty();
    }
}